app/controllers/authenticate_controller.rb

```diff
@@ -68,7 +68,8 @@ def authenticate
     authn_token = Authentication::Authenticate.new.(
       authenticator_input: authenticator_input,
       authenticators: installed_authenticators,
-      enabled_authenticators: Authentication::InstalledAuthenticators.enabled_authenticators_str(ENV)
+      enabled_authenticators: Authentication::InstalledAuthenticators.enabled_authenticators_str(ENV),
+      encoded_response: params[:encoded]
```

Comment:
this can move to the `authenticator_input` object like we do with the rest of the params.

btw, i don't see that you defined this param in `routes.rb`. did you verify that this works? what do you think about adding an integration test?

Reply:
1. Ok.
2. It works. its optional POST parameter - ..../authenticate?encoded=. I didn't see any example in routes.rb to define it.

Reply:
you are correct, i looked again and you don't need to add anything to `routes.rb`